lib/EnvironmentPlugin.js

```diff
@@ -12,15 +12,18 @@ const WebpackError = require("./WebpackError");
 /** @typedef {import("./DefinePlugin").CodeValue} CodeValue */
 
 class EnvironmentPlugin {
+	/**
+	 * @param {(string | string[] | Record<string, string>)[]} keys keys
+	 */
```

Comment:
@alexander-akait Hi there, it seems like this PR broke the types for EnvironmentPlugin. [According to docs](https://webpack.js.org/plugins/environment-plugin/), both undefined and null are accepted as values (and have different behavior), however with this change it is no longer allowed to pass null: ([playground link](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbwKIDsBuwoRSApimABQBsBXAc2BQF84AzLEOAIgHdcAjMAQwGMBrZgG4AUCJS5WcVBiw58RMpRQAKBCLia4AMQDyugFxxSKACa46VXKYA0GrQCEAggCUjKUsWJ3qASiFAA))

<img width="956" alt="image" src="https://github.com/user-attachments/assets/a521b889-b25a-46d5-9c0f-617891921990">

Reply:
filed https://github.com/webpack/webpack/issues/18719

Reply:
PR: #18715